src/ftsh_functions.c

```diff
@@ -1,70 +1,110 @@
 #include "ftsh.h"
 #include "ftsh_functions.h"
-
+#include "search.h"
+#include "match.h"
+#include "../src/simclist.h"
 
 /*
     List of builtin commands, followed by their corresponding functions.
  */
 char *builtin_str[] = {
     "help",
     "exit",
-    "load"
+    "find"
 };
 
 
-int (*builtin_func[]) (char **) = {
+int (*builtin_func[]) (char **, FILE *pf) = {
     &ftsh_help,
     &ftsh_exit,
-    &ftsh_load
+    &ftsh_find
 };
 
 
-int ftsh_help(char **args)
-{
-    printf("Full-Text Shell:\n");
+int ftsh_help(char **args, FILE *pf) {
+    printf("[FULL-TEXT SEARCH]\n");
     printf("Type program names and arguments, and hit enter.\n");
     printf("The following are built in:\n");
 
-    for (int i = 0; i < ftsh_num_builtins(); i++) {
+    int i = 0;
+    for (i = 0; i < ftsh_num_builtins(); i++) {
         printf("  %s\n", builtin_str[i]);
     }
 
-    printf("Use the man command for information on other programs.\n");
     return 1;
 }
 
 
-int ftsh_exit(char **args)
-{
+int ftsh_exit(char **args, FILE *pf) {
     return 0;
 }
 
 
-int ftsh_load(char **args)
-{
-    char ch;
-    char *path = args[1]; // second argument is filepath
-    
-    FILE *fp = fopen(path, "r");
+int ftsh_find(char **args, FILE *pf) {
+    int STATUS = 1;
+    char buf[100];
+    char *input;
+
+    int start_line = 1;
+    int BUFFER_LENGTH = 100;
+    char *word = args[1];
+
+    match curMatch;
+    list_t matches;
+    list_init(&matches);
+
+
+    /* Finding first match at minimum */
+    while (list_size(&matches) == 0 && fgetc(pf) != EOF) {
+        matches = *parse_file_buffered(pf, start_line, \
+                                       (start_line + BUFFER_LENGTH), word, &matches);
+
+        start_line += BUFFER_LENGTH;
+
+        if (fgetc(pf) == EOF && list_size(&matches) == 0) {
+            printf("No matches for %s have been found.\n", word);
+            return 1;
+        }
+    }
+
+    int index = 0;
+    curMatch = *match_get_at_index(index, &matches);
+
+    match_display(&curMatch);
+    list_info(&matches);
+
+    /* NEEDS FIX */
+    while (STATUS) {
```

Comment:
Let's `#define` a couple variables, `SHOULD_CONTINUE` and `SHOULD_EXIT`, so that we can say `while (status == SHOULD_CONTINUE`. Change all the `ftsh` builtins to return those values as well.

Reply:
done!